src/actions/loginUsers.ts

```diff
@@ -6,11 +6,28 @@ import { redirect } from 'next/navigation'
 export const loginUser = async (previousState: unknown, formData: FormData) => {
   const supabase = await supabaseServerClient()
 
-  const email = formData.get('email') as string
+  const identifier = formData.get('email') as string //should change form name
   const password = formData.get('password') as string
 
   let success = true
 
+  let email = identifier
+  if (!identifier.includes('@')) {
```

Comment:
put all this code in its own function, call it something like `getEmailFromUserIdentifier`